cmd/outline-ss-server/config.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+type Service struct {
```

Comment:
I suggest we name the config types with a `Config` suffix to make it clear they are just the config, not the actual objects.

For example, this is not a Service, but a Service config:
```suggestion
type ServiceConfig struct {
```